routing/car_router.cpp

```diff
@@ -282,7 +282,16 @@ bool CarRouter::FindRouteMSMT(TFeatureGraphNodeVec const & sources,
       {
       case NoError: return true;
       case Cancelled: return false;
-      default: continue;
+      case NoCurrentPosition: return false;
```

Comment:
There's only one place where something is called:
```c++
void Framework::BuildRoute(m2::PointD const & finish, uint32_t timeoutSec)
{
  ASSERT_THREAD_CHECKER(m_threadChecker, ("BuildRoute"));
  ASSERT(m_drapeEngine != nullptr, ());

  m2::PointD start;
  if (!m_drapeEngine->GetMyPosition(start))
  {
    CallRouteBuilded(IRouter::NoCurrentPosition, storage::TCountriesVec());
    return;
  }
  BuildRoute(start, finish, false /* isP2P */, timeoutSec);
}
```
So, I guess it's possible to put an ASSERT here.